freud/locality.pyx

```diff
@@ -1120,17 +1120,28 @@ cdef class Voronoi(_Compute):
     def __dealloc__(self):
         del self.thisptr
 
-    def compute(self, system):
+    def compute(self, system, radii=None):
         R"""Compute Voronoi diagram.
 
         Args:
             system:
                 Any object that is a valid argument to
                 :class:`freud.locality.NeighborQuery.from_system`.
+            radii:
+                A list of :class:`numpy.ndarray` defining radii of points in
+                the system. (Default value = :code:`None`, meaning unbiased
+                Voronoi diagram.)
         """
         cdef NeighborQuery nq = NeighborQuery.from_system(system)
-        self.thisptr.compute(nq.get_ptr())
         self._box = nq.box
+        if radii is None:
+            radii = np.zeros(len(nq.points), dtype=np.float64)
+        radii = freud.util._convert_array(
+            radii,
+            shape=(len(nq.points),),
+            dtype=np.float64)
+        cdef double[::1] l_radii = radii
+        self.thisptr.compute(nq.get_ptr(), &l_radii[0])
```

Comment:
If `radii is None`, we want to pass `NULL` instead of an array of zeros. Refer to this example of how to use `None` and pass `NULL` to the C++ side.
https://github.com/glotzerlab/freud/blob/0e695ea11daf5385c546bfa4688c2009064aee93/freud/cluster.pyx#L88-L99
All the `cdef` declarations must be outside of conditionals (outside of `if` statements), which is why we declare `l_radii` before the `if`.
```suggestion
        cdef double* l_radii_ptr = NULL
        cdef double[::1] l_radii
        if radii is not None:
            l_radii = freud.util._convert_array(
                radii, shape=(len(nq.points),), dtype=np.float64)
            l_radii_ptr = &l_radii[0]
        self.thisptr.compute(nq.get_ptr(), l_radii_ptr)
```